vip-helpers/vip-newrelic.php

```diff
@@ -0,0 +1,134 @@
+ */
+function wpcom_vip_add_uri_to_newrelic() {
+	if ( ! is_admin() && function_exists( 'newrelic_add_custom_parameter' ) ) {
+		newrelic_capture_params();
+		newrelic_add_custom_parameter( 'HTTP_REFERER', isset( $_SERVER['HTTP_REFERER'] ) ? $_SERVER['HTTP_REFERER'] : '' );
+		newrelic_add_custom_parameter( 'HTTP_USER_AGENT', isset( $_SERVER['HTTP_USER_AGENT'] ) ? $_SERVER['HTTP_USER_AGENT'] : '' );
+		newrelic_add_custom_parameter( 'HTTPS', is_ssl() );
+	}
+}
+
+/**
+ * Name cron correctly in New Relic and do not count it as part of the Apdex score.
+ *
+ * We don't want to count cron as part of the apdex because it is not a user facing function and if cron tasks are slow it doesn't imply that the site's performance is impacted. Without removing these, long running cron tasks could flag the site as having performance problems, which would cause false positives in the monitoring.
+ */
+function wpcom_vip_cron_for_newrelic() {
+	if ( wp_doing_cron()
+		&& function_exists( 'newrelic_ignore_apdex' )
+		&& function_exists( 'newrelic_add_custom_parameter' )
+		&& function_exists( 'newrelic_name_transaction' ) ) {
+		newrelic_name_transaction( 'wp-cron' );
+		// N.B. VIP Go sites execute cron events through WP CLI, so the event
+		// can be determined in the New Relic 'wp-cli-cmd-args' custom parameter
+		newrelic_ignore_apdex();
+	}
+}
+
+/**
+ * Name wp-cli correctly in New Relic and do not count it as part of the Apdex score
+ *
+ * We don't want to count ongoing WP-CLI requests as part of the apdex because it is not a user facing function and if a WP-CLI request is slow it doesn't imply that the site's performance is impacted. Without removing these WP-CLI requests from the apdex calculation it could flag the site as having performance problems, which would cause false positives in the monitoring.
+ */
+function wpcom_vip_wpcli_for_newrelic() {
+	if ( defined( 'WP_CLI' )
+	     && WP_CLI
+	     && ! wp_doing_cron()  // Cron is going to be run via WP-CLI in the near term. We want to keep Cron and WP-CLI separated for better monitoring so we're not going to flag WP_CLI requests that are actually cron requests as WP-CLI.
+	     && function_exists( 'newrelic_ignore_apdex' )
+	     && function_exists( 'newrelic_add_custom_parameter' )
+	     && function_exists( 'newrelic_name_transaction' ) ) {
+
+		newrelic_name_transaction( 'wp-cli' );
+
+		$wp_cli_arguments = \WP_CLI::get_runner()->arguments;
+		if ( empty( $wp_cli_arguments ) ) {
+			// Not much to do at this point
+			return;
+		}
+		if ( ! is_array( $wp_cli_arguments ) ) {
+			$wp_cli_arguments = [ $wp_cli_arguments ];
+		}
+		array_unshift( $wp_cli_arguments, 'wp' );
+		$cmd = implode( ' ', $wp_cli_arguments );
+		// e.g. `wp option get siteurl`
+		newrelic_add_custom_parameter( 'wp-cli-cmd', $cmd );
+
+		$assoc_args = \WP_CLI::get_runner()->assoc_args;
+		$config = \WP_CLI::get_runner()->config;
+		// Empty values clutter the result
```

Comment:
`array_filter()` can accomplish the same.

Reply:
@simonwheatley Any reason not to use it?

Reply:
@simonwheatley ping about ^^